### PR TITLE
Issue 254 - Apply linting correctly during build

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
And fix accumulated lint errors that were missed.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -5,7 +5,10 @@
     ],
     "linterOptions": {
         "exclude": [
-            "cypress/**"
+            "**/.!{ts,tsx}",
+            "node_modules/**",
+            "cypress/**",
+            "@Types/**"
```
Comment (Marc-Andre-Rivet) [inline tslint.json]: For VSCode. The script does not need this

File: package.json [hunk 2]
```diff
@@ -23,8 +23,8 @@
     "private::wait_dash8082": "wait-on http://localhost:8082",
     "private::wait_dash8083": "wait-on http://localhost:8083",
     "private::wait_js": "wait-on http://localhost:8080",
-    "private::lint.js": "eslint src",
-    "private::lint.ts": "tslint --project . src/**/*.ts",
+    "private::lint.js": "eslint '{src,demo,tests}/**/*.js'",
+    "private::lint.ts": "tslint '{src,demo,tests}/**/*.{ts,tsx}' --exclude '**/@Types/*.*'",
```
Comment (Marc-Andre-Rivet) [inline package.json]: the ```@Types``` folder is special

File: tests/cypress/support/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./../../../.eslintrc",
+  "globals": {
+    "Cypress": true,
+    "cy": true
+  },
+}
```
Comment (Marc-Andre-Rivet) [inline tests/cypress/support/.eslintrc]: Cypress and cy variables are pulled out of thin air in the cypress environment

File: package.json (hunk 2 above)
Comment (valentijnnieman) [reply package.json]: ✨ special ✨

File: tslint.json (hunk 1 above)
Comment (valentijnnieman) [reply tslint.json]: Isn't there a specific vscode config file you can dump this in?

File: dash_table/package-info.json [hunk 4]
```diff
@@ -23,8 +23,8 @@
     "private::wait_dash8082": "wait-on http://localhost:8082",
     "private::wait_dash8083": "wait-on http://localhost:8083",
     "private::wait_js": "wait-on http://localhost:8080",
-    "private::lint.js": "eslint src",
-    "private::lint.ts": "tslint --project . src/**/*.ts",
+    "private::lint.js": "eslint '{src,demo,tests}/**/*.js'",
+    "private::lint.ts": "tslint '{src,demo,tests}/**/*.{ts,tsx}' --exclude '**/@Types/*.*'",
```
Comment (valentijnnieman) [inline dash_table/package-info.json]: Won't tslint run on .js files too? Is it necessary to have both `eslint` and `tslint`?

File: tslint.json (hunk 1 above)
Comment (Marc-Andre-Rivet) [reply tslint.json]: Not in this case. This is picked up by the extension

**Pull request merged**